### ISSUE TITLE
SS3: Member_ForgotPasswordEmail causes double "from" headers

### ISSUE THREAD

**Issue by axllent** [silverstripe/silverstripe-framework#8873]
## Affected Version

SilverStripe 3.*

## Description

When doing a password reset on SilverStripe 3, the `Member_ForgotPasswordEmail` class is causing a second blank `From` email header in the email headers. 

This double `From` header does not comply with RFC 5322, and some mail servers (eg: Gmail) now reject (bounce) those password reset emails with `Our system has detected that this message is 550-5.7.1 not RFC 5322 compliant: 550-5.7.1 Multiple 'From' headers found.`

## Steps to Reproduce

Request a password reset, view email source.

I suspect this is being caused by:

```
protected $from = '';  // setting a blank from address uses the site's default administrator email
```

**Comment by maxime-rainville** [comment]
Just had a go at reproducing this with SS3/SS4. I only saw one header when using [MailCatcher](https://mailcatcher.me/). I looked at the associated EML file as well. Can any one else confirm they are seeing this?

If this is going to be a systemic issues for people sending mail to Gmail from a SilverStripe 3 site, it might be worth fixing even thought SS3 is in limited support right now.

@axllent Can you give us some info on your mail server set up? Are you using the local mail server? Are you using some third party service to send you emails? Do you have any mail specific setting in your config? Maybe try on a different server or with a different mail service and see if that makes any difference.

**Comment by axllent** [comment]
@maxime-rainville The email is being sent directly via postfix on the server (hence the logs above), although the mail is proxied from a docker container (on the same host) to postfix using [msmtp](https://marlam.de/msmtp/).

I can confirm that regular use of the `Email` class does not have this issue, so it seems specific to `Member_ForgotPasswordEmail`. It is possible your SMTP MTA (or in this case MailCatcher) may be dropping the duplicate `From` header... not sure. I also don't see why msmtp would be adding a header for these and not `Email`, so I think that is highly unlikely, and the only difference being that `Member_ForgotPasswordEmail` has a `protected $from = '';` variable.

I'd need to set up a separate environment to test directly with sendmail/postfix which will take some time to test it "natively" (ie: without a proxy), but at this stage I'd say that it's unlikely that is the cause.

**Comment by axllent** [comment]
Actually I just found [this](https://marlam.de/msmtp/msmtp.html#add_005fmissing_005ffrom_005fheader) - `This command controls whether to add a From header if the mail does not have one. The default is to add it.`

So I disabled that feature, and I can confirm that just a single blank "From" header is included with those `Member_ForgotPasswordEmail` emails.

```
Subject: =?UTF-8?B?WW91ciBwYXNzd29yZCByZXNldCBsaW5r?=
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="----=_NextPart_122082010519"
Content-Transfer-Encoding: 7bit
From: 
X-Mailer: SilverStripe Mailer - version 2006.06.21 (Sent from "71.apache.tj.hq")
X-Priority: 3
X-SilverStripeSite: mysite
```

So I think that answers it, though I'm not sure if this is a feature as such. This means that for every site `Email:admin_email` *must* be set for the ForgotPasswordEmail to work, otherwise a blank "From" is sent regardless of the system's default configuration. I don't ever seeing that, however I would say I've probably answered my own question.

Are you happy for me to close, or should this be addressed @maxime-rainville ?
